controllers/clusters/helpers.go

```diff
@@ -202,10 +204,42 @@ func createSpecDifferenceMessage(k8sSpec, iSpec any) (string, error) {
 		return "", err
 	}
 
-	msg := "There are external changes on the Instaclustr console. Please reconcile the specification manually. "
-	specDifference := fmt.Sprintf("k8s spec: %s; data from instaclustr: %s", k8sData, iData)
+	var k8sSpecMap map[string]any
+	err = json.Unmarshal(k8sData, &k8sSpecMap)
+	if err != nil {
+		return "", err
+	}
+
+	var iSpecMap map[string]any
+	err = json.Unmarshal(iData, &iSpecMap)
+	if err != nil {
+		return "", err
+	}
+
+	const externalChangesBaseMsg = "There are external changes on the Instaclustr console. Please reconcile the specification manually."
+
+	diffs := dcomparison.MapsDiff("spec", k8sSpecMap, iSpecMap)
+	if len(diffs) == 1 {
+		return fmt.Sprintf(
+			"%s Diff: {path: %s, k8sValue: %s, instaclustrValue: %s}",
+			externalChangesBaseMsg,
+			diffs[0].Path,
+			diffs[0].Value1,
+			diffs[0].Value2,
+		), nil
+	}
+
+	var diffMessages []string
+	for _, diff := range diffs {
+		diffMessages = append(diffMessages, fmt.Sprintf(
+			"{field: %s, k8sValue: %v, instaclustrValue: %v}",
+			diff.Path,
+			diff.Value1,
+			diff.Value2,
+		))
+	}
```

Comment:
There are a lot of cases when our specification differs from instaclustr API specification. Does that mean that in all such cases even if the values are equal, but the structure is different, we will receive such messages about external changes?

Reply:
Before it, we received the spec from the Instaclustr API and unmarshaled it to the specific struct (e.g we got cassandra cluster status and unmarshal its spec to the `Cassandra`). Also, before calling `MapsDiff` we deleted redundant fields from the spec as `userReferences`. We already don't have any differences between Instaclustr and k8s representation of resources specs.